File: Algorithms Techniques/2. Searching/Linear-Search/linearSearch.cpp
#include <iostream>
#include <vector>
using namespace std;

int linearSearch(vector<int> array, int target) {
  // searching element linearly -- O(n)
  for (int i = 0; i < array.size(); i++)
    if (array[i] == target) return i; 
  // if can't find it
  return -1;
}

int main() {
  cout << "Create array {1, 3, 8, 2, 9, 2, 5, 6}" << endl;
  vector<int> array{1, 3, 8, 2, 9, 2, 5, 6};
  int size = 8;
  int index = linearSearch(array, 3);
  if (index < 0) 
    cout << "Can't find it!" << endl;
  else 
    cout << "Found it at index: " << index << endl;
}

File: Algorithms Techniques/2. Searching/Linear-Search/linearSearch.py
# linear search definition
def linear_search(array, target):
  for i in range(len(array)):
    if array[i] == target:
      return i # found it, return index
  return -1 # can't find it

# formatted linear search function output
def search_output(index):
  if index == -1:
    print("Given value is not in the array!")
  else:
    print("Found it! It's at index " + str(index))

#==============
#   TESTING   #
#==============

# create array
array = [1, 3, 8, 2, 9, 2, 5, 6]
print("- Created array with value [1, 3, 8, 2, 9, 2, 5, 6]")
print

# testing the function
print("- Searching for 2")
search_output(linear_search(array, 2))
print

print("- Searching for 10")
search_output(linear_search(array, 10))